tests/system/test_session_api.py

```diff
@@ -180,12 +178,33 @@
     PostGresAllTypesRowData(pkey=101, int_value=123),
     PostGresAllTypesRowData(pkey=102, bool_value=False),
     PostGresAllTypesRowData(pkey=103, bytes_value=BYTES_1),
+    PostGresAllTypesRowData(pkey=104, date_value=SOME_DATE),
     PostGresAllTypesRowData(pkey=105, float_value=1.4142136),
     PostGresAllTypesRowData(pkey=106, string_value="VALUE"),
     PostGresAllTypesRowData(pkey=107, timestamp_value=SOME_TIME),
     PostGresAllTypesRowData(pkey=108, timestamp_value=NANO_TIME),
     PostGresAllTypesRowData(pkey=109, numeric_value=NUMERIC_1),
     PostGresAllTypesRowData(pkey=110, jsonb_value=JSON_1),
+    # empty array values
+    PostGresAllTypesRowData(pkey=201, int_array=[]),
+    PostGresAllTypesRowData(pkey=202, bool_array=[]),
+    PostGresAllTypesRowData(pkey=203, bytes_array=[]),
+    PostGresAllTypesRowData(pkey=204, date_array=[]),
+    PostGresAllTypesRowData(pkey=205, float_array=[]),
+    PostGresAllTypesRowData(pkey=206, string_array=[]),
+    PostGresAllTypesRowData(pkey=207, timestamp_array=[]),
+    PostGresAllTypesRowData(pkey=208, numeric_array=[]),
+    PostGresAllTypesRowData(pkey=209, jsonb_array=[]),
+    # non-empty array values, including nulls
+    PostGresAllTypesRowData(pkey=301, int_array=[123, 456, None]),
+    PostGresAllTypesRowData(pkey=302, bool_array=[True, False, None]),
+    PostGresAllTypesRowData(pkey=303, bytes_array=[BYTES_1, BYTES_2, None]),
+    PostGresAllTypesRowData(pkey=304, date_array=[SOME_DATE, None]),
```

Comment:
Should we add another date here, similar to how we have for other arrays?
```suggestion
    PostGresAllTypesRowData(pkey=304, date_array=[SOME_DATE, SOME_DATE, None]),
```